code/_onclick/hud/new_player.dm

```diff
@@ -348,14 +369,139 @@
+		collapse_button()
+	else
+		menu_shutter.expand_bottom_buttons()
+		expand_button()
+
+///Moves the button to the top of the screen, leaving only the screen part in view
+/atom/movable/screen/lobby/button/collapse/proc/collapse_button()
+	ce_button_matrix = matrix()
+	//wait for the shutter to come down
+	animate(src, transform = ce_button_matrix, time = SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION)
+	//then pull the button up with the shutter and leave it on the edge of the screen
+	animate(transform = ce_button_matrix.Translate(x = 0, y = 134), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_IN)
+
+///Extends the button back to its usual spot
+/atom/movable/screen/lobby/button/collapse/proc/expand_button()
+	animate(src, transform = ce_button_matrix.Translate(x = 0, y = -134), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_OUT)
+	ce_button_matrix = matrix()
+
+///Disables/enables menu buttons on the client's screen
+/atom/movable/screen/lobby/button/collapse/proc/toggle_menu(mob/dead/new_player/new_player)
+	if(hud.inventory_shown)
+		new_player.client.screen -= hud.toggleable_inventory
+	else
+		new_player.client.screen += hud.toggleable_inventory
+	hud.inventory_shown = !hud.inventory_shown
+
+/atom/movable/screen/lobby/shutter
+	icon = 'icons/hud/lobby/shutter.dmi'
+	icon_state = "shutter"
+	base_icon_state = "shutter"
+	screen_loc = "TOP,CENTER:-73"
+	layer = LOBBY_SHUTTER_LAYER
+	always_shown = TRUE
+	///List of bottom buttons on the UI; these are animated such as they appear to attach to the shutter
+	///when it moves
+	var/list/bottom_buttons = list()
+	var/matrix/shutter_matrix = matrix()
+
+/atom/movable/screen/lobby/shutter/SlowInit()
+	. = ..()
+	//screen_loc is initialized right above the main UI(the big three buttons), so we use
+	//matrix transform to shift it up to be off-screen
+	shutter_matrix.Translate(x = 0, y = 143) //the shutter moves just a wee bit shorter than the bottom buttons
+	transform = shutter_matrix
+
+///This animates moving the bottom buttons off-screen, disabling them
+/atom/movable/screen/lobby/shutter/proc/collapse_bottom_buttons()
+	for(var/atom/movable/screen/lobby/button/bottom/button_to_scroll_up in bottom_buttons)
+		button_to_scroll_up.button_matrix = matrix()
+		//wait for the shutter to come down
+		animate(button_to_scroll_up, transform = button_to_scroll_up.button_matrix, time = SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION)
+		//then pull the buttons up with the shutter
+		animate(transform = button_to_scroll_up.button_matrix.Translate(x = 0, y = 146), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_IN)
+		if(istype(button_to_scroll_up, /atom/movable/screen/lobby/button/bottom/poll))
+			var/atom/movable/screen/lobby/button/bottom/poll/poll_button = button_to_scroll_up
+			if(!poll_button.new_poll) //don't deactivate the poll button unless a poll is up (it's already inactive)
+				continue
+		button_to_scroll_up.set_button_status(FALSE)
```

Comment:
part of me thinks this should be handled with a signal rather then a loop like this

Reply:
yeah that sounds saner, will do that

Reply:
https://github.com/tgstation/tgstation/commit/28ffee160e7e3ecd6602535f6400597692c0b938